### PR TITLE
GithubService::getRepositoryLicense tests

### PR DESCRIPTION
**Pull request by namefreezers**
# Description

Added unit test set (4 pcs) for the GithubService::getRepositoryLicense method. This step is required for the GitHub action infra setup.
Refactored creation of `GitHubService` in order to simplify unit testing.

New dependency:
`mockito-inline:4.7.0`

Needed for mock external library objects. `mockito-inline` instead of `mockito-core` is used, because only `mockito-inline` supports mocking static methods

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Code cleanup/refactoring
- [ ] Documentation update
- [ ] This change requires a documentation update
- [ ] CI system update
- [x] Test Coverage update

# How Has This Been Tested?

**Test Configuration**:
* Java: v18
* LPVS Release: v1.x.x

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by namefreezers** [comment]
Please carefully review my changes to [src/main/java/com/lpvs/service/GitHubService.java](https://github.com/Samsung/LPVS/pull/19/files#diff-c788a0a699a9d423378bc544766e7a9bb989247c0db6f8ef4a999ea9e3828609)

Thank you!

File: src/test/java/com/lpvs/service/GitHubServiceTest.java
Comment (MoonkiHong) [inline src/test/java/com/lpvs/service/GitHubServiceTest.java:1]: Copyright statement should be included at the top of every single java file. Please refer to the case here. [src/main/java/com/lpvs/service/GitHubService.java](https://github.com/Samsung/LPVS/pull/19/files#diff-c788a0a699a9d423378bc544766e7a9bb989247c0db6f8ef4a999ea9e3828609)

**Review by MoonkiHong — CHANGES_REQUESTED** [review]
Please take a look at my comment. 😄

File: src/test/java/com/lpvs/service/GitHubServiceTest.java
Comment (o-kopysov) [inline src/test/java/com/lpvs/service/GitHubServiceTest.java:249]: Why do you leave a commented-out code?

**Review by o-kopysov — CHANGES_REQUESTED** [review]

File: src/test/java/com/lpvs/service/GitHubServiceTest.java
Comment (namefreezers) [reply src/test/java/com/lpvs/service/GitHubServiceTest.java:249]: Commented code was left intentionally. To show the difference between test cases. Intention was to show difference between `LicensePresent` and `LicenseAbsent` test cases in this case. 

Should I delete it?

File: src/test/java/com/lpvs/service/GitHubServiceTest.java
Comment (namefreezers) [reply src/test/java/com/lpvs/service/GitHubServiceTest.java:8]: Thank you for remark. 
Copyright statement was added.

File: src/test/java/com/lpvs/service/GitHubServiceTest.java
Comment (MoonkiHong) [reply src/test/java/com/lpvs/service/GitHubServiceTest.java:256]: Resolved. I will wait for the discussion below, then approve it. 😄

File: src/test/java/com/lpvs/service/GitHubServiceTest.java
Comment (MoonkiHong) [reply src/test/java/com/lpvs/service/GitHubServiceTest.java:8]: Resolved my comment. I will approve this PR after the discussion below. 😄

File: src/test/java/com/lpvs/service/GitHubServiceTest.java
Comment (o-kopysov) [reply src/test/java/com/lpvs/service/GitHubServiceTest.java:256]: I agree, please keep this commented code as is.

**Review by o-kopysov — APPROVED** [review]
LGTM

**Review by MoonkiHong — APPROVED** [review]
+1

**Pull request merged**
